src/diem/jsonrpc/client.py

```diff
@@ -399,6 +399,10 @@ def get_diem_id_domain_map(self, batch_size: int = 100) -> typing.Dict[str, str]
             event_index += batch_size
         return domain_map
 
+    def support_diem_id(self) -> bool:
+        tc_account = self.must_get_account(TREASURY_ADDRESS)
+        return True if tc_account.role.diem_id_domain_events_key else False
```

Comment:
nit: afaik this could just be `return bool(tc_account.role.diem_id_domain_events_key)`